wordpress-mobile/WordPress-Android
# Feature/15360 add tracks for domains

Fixes #15360 

Adds the following tracks for domains

- domains_dashboard_viewed
- domains_dashboard_get_domain_tapped
- domains_dashboard_add_domain_tapped
- domains_dashboard_select_domain_tapped
- domains_registration_form_viewed
- domains_registration_form_submitted
- domains_purchase_webview_viewed
- domains_purchase_domain_success

To test:

**Setup: Enable feature flag**

1. On the Main screen, tap the Me button.
1. On the Me screen, go to App Settings > Debug settings.
1. On the Debug Settings screen, scroll down to the "Features in development" section and tap `SiteDomainsFeatureConfig` to enable the Domains feature flag.

**Test**

1. Open the app and select a site.
1. On the Main screen, tap the "Domains" item under the "Configuration" section.
1. On the Domains Dashboard screen.
1. Tap on various buttons, move to search, checkout and purchase
1. Test with a site that has domain credit, doesn't have domain credit... Observe the events in Logcat on AS

```
🔵 Tracked: domains_dashboard_viewed
🔵 Tracked: domains_dashboard_get_domain_tapped
🔵 Tracked: domain_credit_redemption_tapped
🔵 Tracked: domains_dashboard_select_domain_tapped
🔵 Tracked: domains_dashboard_add_domain_tapped
🔵 Tracked: domains_registration_form_viewed
🔵 Tracked: domains_purchase_webview_viewed
🔵 Tracked: domains_purchase_domain_success

```

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
